src/dft/backends/rocfft/commit.cpp

```diff
@@ -259,12 +260,37 @@ class rocfft_commit final : public dft::detail::commit_impl<prec, dom> {
         std::reverse(stride_vecs.vec_b.begin(), stride_vecs.vec_b.end());
         stride_vecs.vec_b.pop_back(); // Offset is not included.
 
-        rocfft_plan_description plan_desc;
-        if (rocfft_plan_description_create(&plan_desc) != rocfft_status_success) {
+        // This workaround is needed due to a confirmed issue in rocFFT from version
+        // 1.0.23 to 1.0.30. Those rocFFT version correspond to rocm version from
+        // 5.6.0 to 6.3.0.
+        // Link to rocFFT issue: https://github.com/ROCm/rocFFT/issues/507
+        if constexpr (rocfft_version_major == 1 && rocfft_version_minor == 0 &&
+                      (rocfft_version_patch > 22 && rocfft_version_patch < 31)) {
+            if (dom == dft::domain::COMPLEX && dimensions > 2) {
+                auto stride_checker = [&](const auto& a, const auto& b) {
+                    for (ulong i = 0; i < dimensions; ++i) {
+                        if (a[i] != b[i])
+                            return false;
+                    }
+                    return true;
+                };
+                std::printf("hello\n");
+                if (!stride_checker(stride_vecs.vec_a, stride_vecs.vec_b))
+                    throw oneapi::mkl::unimplemented(
+                        "DFT", func,
+                        "due to a bug in rocfft version in use, it requires fwd and bwd stride to be the same for COMPLEX out_of_place computations");
```

Comment:
I think this whole check should be run in the out-of-place compute functions otherwise it looks like this will prevent users from running complex in-place FFTs for no reasons.

Reply:
yes, that would make more sense, but we can let that function clean and I can add another check in the `if` at [line 269](https://github.com/oneapi-src/oneMKL/blob/54e0cf91910a6584e8c81c522f0a2417cb7a6e10/src/dft/backends/rocfft/commit.cpp#L269). I added a comment directly to that line.

Reply:
You're right, I forgot that the placement was already known at commit time so that makes. Btw you should make sure the changes are formatted with clang-format-9 to avoid issues in the CI!